From 411064cecb6b0c86ab62794011e8680a696f3c32 Mon Sep 17 00:00:00 2001
From: Yelena <yelena@marvell.com>
Date: Wed, 9 Apr 2014 15:51:59 +0300
Subject: [PATCH 1549/1825] pp3: check HMAC queue status before representation

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit b341342ac9afe06fe6ae8b933d893de31e316b9d

	Define maximum frame number to 4 frame

Change-Id: I7c3aba53167bfbc5b6295c837015476aba1f74cd
Signed-off-by: Yelena <yelena@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/7044
Reviewed-by: Uri Eliyahu <uriel@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Dmitri Epshtein <dima@marvell.com>
Tested-by: Dmitri Epshtein <dima@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 drivers/net/ethernet/marvell/pp3/hmac/mv_hmac.c |    6 ++++++
 drivers/net/ethernet/marvell/pp3/hmac/mv_hmac.h |    2 +-
 2 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/drivers/net/ethernet/marvell/pp3/hmac/mv_hmac.c b/drivers/net/ethernet/marvell/pp3/hmac/mv_hmac.c
index 62d54d1..d3b2f4f 100644
--- a/drivers/net/ethernet/marvell/pp3/hmac/mv_hmac.c
+++ b/drivers/net/ethernet/marvell/pp3/hmac/mv_hmac.c
@@ -439,12 +439,18 @@ static void mv_pp3_hmac_queue_show(struct mv_pp3_hmac_queue_ctrl *rxq)
 
 void mv_pp3_hmac_rx_queue_show(int frame, int queue)
 {
+	if (!mv_hmac_rxq_handle[frame][queue])
+		return;
+
 	pr_info("\n-------------- HMAC RX queue (0x%p) -----------", mv_hmac_rxq_handle[frame][queue]);
 	return mv_pp3_hmac_queue_show(mv_hmac_rxq_handle[frame][queue]);
 }
 
 void mv_pp3_hmac_tx_queue_show(int frame, int queue)
 {
+	if (!mv_hmac_txq_handle[frame][queue])
+		return;
+
 	pr_info("\n-------------- HMAC TX queue (0x%p) -----------", mv_hmac_txq_handle[frame][queue]);
 	return mv_pp3_hmac_queue_show(mv_hmac_txq_handle[frame][queue]);
 }
diff --git a/drivers/net/ethernet/marvell/pp3/hmac/mv_hmac.h b/drivers/net/ethernet/marvell/pp3/hmac/mv_hmac.h
index b8fc589..b6bf2d7 100644
--- a/drivers/net/ethernet/marvell/pp3/hmac/mv_hmac.h
+++ b/drivers/net/ethernet/marvell/pp3/hmac/mv_hmac.h
@@ -70,7 +70,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "common/mv_hw_if.h"
 #include "hmac/mv_hmac_regs.h"
 
-#define MV_PP3_HMAC_MAX_FRAME			(16)
+#define MV_PP3_HMAC_MAX_FRAME			(4)
 #define MV_PP3_QUEUES_PER_FRAME			(16)
 
 #define MV_PP3_HMAC_DG_SIZE				(16)	/* bytes */
-- 
1.7.5.4

